go.mod

```diff
@@ -14,6 +14,7 @@ require (
 	github.com/google/go-cmp v0.5.0
 	github.com/google/gofuzz v1.1.0
 	github.com/google/uuid v1.1.1
+	github.com/hashicorp/go-retryablehttp v0.6.6
```

Comment:
seems like a cool library

Reply:
It looked like the appropriate choice after a bit of research. Hashicorp is usually high-quality, and CAPZ already has some of their other libraries vendored in go.sum.

I like to keep vendoring as minimal as possible, so I also thought about writing our own retry loop. This can be surprisingly tricky--the retryablehttp docs convinced me I would probably miss some edge cases. But I'm happy to write the retry loop and avoid vendoring if that's a better tradeoff.